Sources/JobsRedis/RedisJobQueue.swift

```diff
@@ -24,31 +25,47 @@ import RediStack
 public final class RedisJobQueue: JobQueueDriver {
     public struct JobID: Sendable, CustomStringConvertible {
         let id: String
+        let delayUntil: Int64
 
-        public init() {
+        public init(delayUntil: Double?) {
             self.id = UUID().uuidString
+            self.delayUntil = Self.toMillis(value: delayUntil)
         }
 
         /// Initialize JobID from String
         /// - Parameter value: string value
         public init(_ value: String) {
-            self.id = value
+            let parts = value.components(separatedBy: ":")
+            self.id = parts[0]
+            self.delayUntil = if parts.count > 1 {
+                Self.toMillisFromString(value: parts[1])
+            } else {
+                0
+            }
         }
 
-        public init(from decoder: Decoder) throws {
-            let container = try decoder.singleValueContainer()
-            self.id = try container.decode(String.self)
+        static func toMillis(value: Double?) -> Int64 {
+            if let value {
+                return Int64(value * 1000)
```

Comment:
You check for overflow
```suggestion
                return Double(value * 1000) < Double(Int64.max) ? Int64(value * 1000) : Int64.max
```